### ISSUE TITLE
Cannot run CLI commands with empty cache due to missing Extend\Entity\EX_OroLocaleBundle_Localization class

### ISSUE THREAD

**Issue by LRotherfield** [oroinc/platform#898]
Hi Guys

I just upgraded from 3.1.0-rc to 3.1.4 and I am unable to run any console commands (that don't use --force) when I have an **empty** cache directory.  I have tried running `cache:clear`, `oro:entity-extend:cache:warmup --env=prod` etc but none of them work.  If I copy the oro_entities extend directory from another environment's cache then it works fine.  It looks like this was introduced by [this commit](https://github.com/oroinc/platform/commit/9855de6136e261a0f3449b30ace3b0bd8da21146) that causes MetaData requests to be made for the Localization entity by a command listener.   

Is this a bug anyone else is seeing?  I can confirm that when the error below is thrown the class is really missing and if I create the class (copying from another env) then the error is not thrown and the commands work as usual.

Please let me know if you need any further information

```php 
php bin/console oro:entity-extend:cache:warmup --env=prod -vvv

In RuntimeReflectionService.php line 63:
                                                                      
  [ReflectionException]                                               
  Class Extend\Entity\EX_OroLocaleBundle_Localization does not exist  
                                                                      

Exception trace:
 () at /vagrant/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/RuntimeReflectionService.php:63
 ReflectionProperty->__construct() at /vagrant/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/RuntimeReflectionService.php:63
 Doctrine\Common\Persistence\Mapping\RuntimeReflectionService->getAccessibleProperty() at /vagrant/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php:963
 Doctrine\ORM\Mapping\ClassMetadataInfo->wakeupReflection() at /vagrant/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:747
 Doctrine\ORM\Mapping\ClassMetadataFactory->wakeupReflection() at /vagrant/vendor/doctrine/persistence/lib/Doctrine/Common/Persistence/Mapping/AbstractClassMetadataFactory.php:181
 Doctrine\Common\Persistence\Mapping\AbstractClassMetadataFactory->getMetadataFor() at /vagrant/vendor/oro/platform/src/Oro/Bundle/EntityBundle/ORM/OroClassMetadataFactory.php:117
 Oro\Bundle\EntityBundle\ORM\OroClassMetadataFactory->getMetadataFor() at /vagrant/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:283
 Doctrine\ORM\EntityManager->getClassMetadata() at /vagrant/vendor/oro/platform/src/Oro/Bundle/EntityBundle/ORM/Repository/EntityRepositoryFactory.php:48
 Oro\Bundle\EntityBundle\ORM\Repository\EntityRepositoryFactory->getRepository() at /vagrant/vendor/doctrine/orm/lib/Doctrine/ORM/EntityManager.php:713
 Doctrine\ORM\EntityManager->getRepository() at /vagrant/vendor/oro/platform/src/Oro/Bundle/EntityBundle/ORM/DoctrineHelper.php:370
 Oro\Bundle\EntityBundle\ORM\DoctrineHelper->getEntityRepositoryForClass() at /vagrant/vendor/oro/platform/src/Oro/Bundle/LocaleBundle/Manager/LocalizationManager.php:193
 Oro\Bundle\LocaleBundle\Manager\LocalizationManager->getRepository() at /vagrant/vendor/oro/platform/src/Oro/Bundle/LocaleBundle/Manager/LocalizationManager.php:96
 Oro\Bundle\LocaleBundle\Manager\LocalizationManager->getLocalizationData() at /vagrant/vendor/oro/platform/src/Oro/Bundle/LocaleBundle/Model/LocaleSettings.php:235
 Oro\Bundle\LocaleBundle\Model\LocaleSettings->getLocale() at /vagrant/vendor/oro/platform/src/Oro/Bundle/LocaleBundle/EventListener/LocaleListener.php:113
 Oro\Bundle\LocaleBundle\EventListener\LocaleListener->onConsoleCommand() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:212
 Symfony\Component\EventDispatcher\EventDispatcher->doDispatch() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/EventDispatcher/EventDispatcher.php:44
 Symfony\Component\EventDispatcher\EventDispatcher->dispatch() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:968
 Symfony\Component\Console\Application->doRunCommand() at /vagrant/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:86
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:248
 Symfony\Component\Console\Application->doRun() at /vagrant/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Console/Application.php:74
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /vagrant/vendor/symfony/symfony/src/Symfony/Component/Console/Application.php:148
 Symfony\Component\Console\Application->run() at /vagrant/bin/console:29
```

**Comment by mbessolov** [comment]
@LRotherfield please specify the application (platform-application, crm-application, commerce-applicatop, or something entirely custom that uses oro/platform package).
Have you been able to reproduce this problem on a vanilla non-customization application?

**Comment by LRotherfield** [comment]
@mbessolov I am using the platform application, there is a bit of custom code although nothing to do with the locale or extend entity bundle.  I have not tried a vanilla version of the platform though.

**Comment by LRotherfield** [comment]
@manowark I don't know if you have managed to replicate this or not but a bit of further testing seems to show it is something cached in redis causing the issue for me:

Redis settings in parameters.yml
```yaml
    session_handler:    'snc_redis.session.handler'
    redis_dsn_session:  'redis://127.0.0.1:6379/0'
    redis_dsn_cache:    'redis://127.0.0.1:6379/1'
    redis_dsn_doctrine: 'redis://127.0.0.1:6379/2'
    redis_setup:        'standalone'
```

On initial installation when redis is empty I can successfully run:

```
php bin/console cache:warmup --env=prod
```

I can then replicate the issue after this first warmup by running:

```
php bin/console cache:clear --no-warmup --env=prod
php bin/console cache:warmup --env=prod
```

This can then be fixed by clearing redis and rerunning the warmup command:
```
redis-cli -p 6379 flushall
php bin/console cache:warmup --env=prod
```

If I run `cache:clear` without the `--no-warmup` option and the cache was populated before running it then nothing breaks.

I hope that helps - if I get time I will try and run this on a fresh install of the platform-application repo

**Comment by LRotherfield** [comment]
Hi @manowark @mbessolov I booted up a fresh install of the platform application on tag 3.1.4 using the vagrant box you provide with it.  I installed redis and then used the same settings as my above post and triggered the same error.  Below are the first commands I ran on the application after the vagrant box booted:
```
[root@localhost oroapp]# php bin/console cache:clear --env=prod

 // Clearing the cache for the prod environment with debug false                                                        

                                                                                                                        
 [OK] Cache for the "prod" environment (debug=false) was successfully cleared.                                          
                                                                                                                        

[root@localhost oroapp]# php bin/console cache:clear --env=prod --no-warmup

 // Clearing the cache for the prod environment with debug false                                                        

                                                                                                                        
 [OK] Cache for the "prod" environment (debug=false) was successfully cleared.                                          
                                                                                                                        

[root@localhost oroapp]# php bin/console cache:warmup --env=prod

In OroEntityExtendBundle.php line 130:
                                                                                                                                                                                                                                        
  In RuntimeReflectionService.php line 63:                                                                                                                                                                                              
                                                                                                                                                                                                                                        
    Class Extend\Entity\EX_OroLocaleBundle_Localization does not exist                                                                                                                                                                  
                                                                                                                                                                                                                                        
                                                                                                                                                                                                                                        
  oro:entity-extend:cache:check [--cache-dir CACHE-DIR] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--disabled-listeners DISABLED-LISTENERS]  
   [--current-user CURRENT-USER] [--current-organization CURRENT-ORGANIZATION] [--] <command>                                                                                                                                           
                                                                                                                                                                                                                                        
```

**Comment by mbessolov** [comment]
@LRotherfield  thank you for providing the details. We will look into it.

**Comment by yshyshkin** [comment]
@LRotherfield here is a hot fix that should help you in this scenario, we will fix it in product ASAP.

```
Index: vendor/oro/platform/src/Oro/Bundle/LocaleBundle/EventListener/LocaleListener.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- vendor/oro/platform/src/Oro/Bundle/LocaleBundle/EventListener/LocaleListener.php	(revision 48e1e13c51256476e26b5c64c0fa41b55d07558b)
+++ vendor/oro/platform/src/Oro/Bundle/LocaleBundle/EventListener/LocaleListener.php	(date 1554480925000)
@@ -112,7 +112,7 @@
             try {
                 $locale   = $this->getLocaleSettings()->getLocale();
                 $language = $this->getLocaleSettings()->getLanguage();
-            } catch (DBALException $exception) {
+            } catch (\Exception $exception) {
                 // application is not installed
                 return;
             }
```

**Comment by anyt** [comment]
Fixed in 3.1.7. Thank you
